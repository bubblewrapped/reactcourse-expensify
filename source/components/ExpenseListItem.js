import React from 'react';
import {Link} from 'react-router-dom';

//{dispatch, id, description, amount, createdAt} is the props variable, remeber it
const ExpenseListItem = ({id, description, amount, createdAt}) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h3>{description}</h3>
		</Link>
		<p>{amount} {createdAt}</p>
	</div>
);

export default ExpenseListItem;