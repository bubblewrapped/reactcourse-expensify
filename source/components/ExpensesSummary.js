import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import selectExpensesTotal from '../selectors/expenses-total.js';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
	return (
		<div className="page-header">
			<div className="content-container">
				<h2 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h2>
				<div className="page-header__actions">
					<Link to="/create" className="button">Add Expense</Link>
				</div>
			</div>
		</div>
	);
};

const mapStateToProps = (state) => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);
	return {
		expenseCount: visibleExpenses.length, 
		expensesTotal: selectExpensesTotal(visibleExpenses)
	};
};

/*export const ExpensesSummary = (props) => (
	<div>
		<p>Viewing {props.expenses.length} {props.expenses.length === 1 ? 'expense' : 'expenses'} totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}</p>
	</div>
);

const mapStateToProps = (state) => {
	return {
		expenses: selectExpenses(state.expenses, state.filters)
	};
};*/

export default connect(mapStateToProps)(ExpensesSummary);