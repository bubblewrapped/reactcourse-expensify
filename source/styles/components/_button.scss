.button {
	border: none;
	padding: $s-size;
	color: white;
	display: inline-block;
	background: $blue;
	font-weight: 300;
	font-size: $font-size-large;
	text-decoration: none;
	line-height: 1;
}

.button--link {
	background: none;
}

.button--secondary {
	background: #888;
}