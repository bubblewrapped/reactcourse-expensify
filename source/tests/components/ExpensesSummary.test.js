import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';

test('Should render ExpensesSummary correctly with many expenses', () => {
	const wrapper = shallow(<ExpensesSummary  expenseCount={3} expensesTotal={12345} />);
	expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary correctly with 1 expense', () => {
	const wrapper = shallow(<ExpensesSummary  expenseCount={1} expensesTotal={1234} />);
	expect(wrapper).toMatchSnapshot();
});