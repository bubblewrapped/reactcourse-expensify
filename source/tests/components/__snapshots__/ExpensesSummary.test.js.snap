// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $12.34
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render ExpensesSummary correctly with many expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $123.45
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
